xija/component/heat.py

```diff
@@ -265,6 +287,37 @@ def plot_solar_heat__pitch(self, fig, ax):
             ax.grid()
 
 
+class NewSolarHeat(SolarHeat, SolarHeatBase):
```

Comment:
I'm having trouble deciding on a new name for this. @taldcroft @matthewdahmer @jeanconn any suggestions? The only "new" thing it would incorporate is the change to the calculation of the yearly variation.

Reply:
Can you just do this like so:
```
class SolarHeatMultiplicative(SolarHeat):
     # Include the new t_phase property and h_phase method
     # Include the override of dvals
```
When I look at this now it is confusing to add the phase stuff in a new base class that is not used anywhere else (in particular it is not the base for `SolarHeat`).

Reply:
I didn't check if the new `t_phase` property is exactly the same as what was getting set in `SolarHeat.dvals`, but if so then it could make sense to add the `t_phase` property to `SolarHeat`, modify that `dvals` to use it, and then just have `SolarHeatMultiplicative` add the `h_phase` and override `dvals`.  In fact there should be just one line of difference, so you could factor that statement out to a `_calc_dvals()` method:
```
def _calc_dvals(self, P_vals, dP_vals):
    # This is overridden by the multiplicative version in SolarHeatMultiplicative
    out =(P_vals + dP_vals * self.var_func(self.t_days, self.tau)
             + self.ampl * np.cos(self.t_phase)).reshape(-1)
    return out
```